[Work-in-Progress] Enables markdown for PageElement text (#80)

Deep-Chill opened the pull request:
- Added content_format to PageElement model
- Added content_format to PageElementSerializer, which checks if text is Markdown or HTML, and then uses the HTMLField to "sanitize" the field.

Deep-Chill committed:
Enables markdown for PageElement text

smirolo commented on pages/serializers.py:
```diff
@@ -185,10 +185,10 @@ class PageElementSerializer(serializers.ModelSerializer):
-        html_tags=settings.ALLOWED_TAGS,
-        html_attributes=settings.ALLOWED_ATTRIBUTES,
-        help_text=_("Long description of the page element"))
+    content_format = serializers.ChoiceField(
```
I am wondering if it is possible to guess the `content_format` from the `text` itself. Maybe not, but that would be awesome.

smirolo commented on pages/serializers.py:
```diff
@@ -255,6 +255,17 @@ def get_path(self, obj):
         return prefix + slug
 
+    @staticmethod
+    def get_text(obj):
```
The `content_format` to HTML conversion is done server-side. What happens if we want to edit the `PageElement.text` as Markdown?
Since the Update Serializer is passed a `content_format` and `text` field (formatted based on the content_format), it could make sense to do the same on the Retrieve API. The code here could be used to add an `html_formatted` read-only field for front-end client convienience.

smirolo reviewed (changes requested):
The Update/Retrieve `text` field should have symmetric semantic.

Deep-Chill committed:
Code refactor

- Use internal_value of the serializer to sanitize the HTML data
- Add a html_formatted field to return text as HTML.

smirolo reviewed (approved):
Looks good. Thanks.
